skopt/tree_opt.py

```diff
@@ -51,11 +54,27 @@ def _tree_minimize(func, dimensions, base_estimator, maxiter,
         # has zero gradient over large distances. As a result we can not
         # use gradient based optimisers like BFGS, so using random sampling
         # for the moment.
-        X = space.transform(space.rvs(n_samples=n_points,
-                                      random_state=rng))
-        values = -gaussian_ei(X, rgr, np.min(yi))
-        next_x = X[np.argmin(values)]
-
+        curr_min = np.min(yi)
+        if acq_min == "sampling":
+            X = space.transform(space.rvs(n_samples=n_points,
+                                          random_state=rng))
+            values = -gaussian_ei(X, rgr, curr_min)
+            next_x = X[np.argmin(values)]
+        elif acq_min == "paramils":
+            for dimension in space.dimensions:
+                if isinstance(dimension, Categorical):
```

Comment:
Sorry for not being clear about what I mean about the API has to be changed. Here are my thoughts:

Assuming some-day we do have estimators that support categorical parameters natively (hopefully GP's soon) and estimators that do not, we would have to make sure that if `dimensions` contains `Categorical`, `space.transform` can return the categorical variables for the estimators that can have categorical support and one-hot encoded space for those that don't. That is what I meant by optionally transform.

Now why, I bring this here is that for categorical variables, in paramils_smac the `n_neighbors` for a given category are the other categories apart from the current given category. Because of this I would like the original categorical space and not the transformed one-hot encoded space.

(I can of course hack up a solution but I would prefer the Space API to do it directly for me)